Add optional min_length to pad_sequence

Darktex opened the pull request:
Summary: Small quality of life change: sometimes you want to pad your input to at least a certain size.

Differential Revision: D10139793

apaszke commented on torch/nn/utils/rnn.py:
```diff
@@ -234,7 +235,10 @@ def pad_sequence(sequences, batch_first=False, padding_value=0):
     max_size = sequences[0].size()
     trailing_dims = max_size[1:]
-    max_len = max([s.size(0) for s in sequences])
+    max_len = max((s.size(0) for s in sequences))
```
I'm pretty sure this doesn't handle the `batch_first=True` case correctly, although it looks like our current code is broken too? Can you please fix that?

apaszke commented on torch/nn/utils/rnn.py:
```diff
@@ -234,7 +235,10 @@ def pad_sequence(sequences, batch_first=False, padding_value=0):
     trailing_dims = max_size[1:]
-    max_len = max([s.size(0) for s in sequences])
+    max_len = max((s.size(0) for s in sequences))
+    if min_length:
```
This conditional is unnecessary. Since the default is `0`, and it's an unsigned values, then `max(x, 0) == x`, so you can always apply it.

Darktex committed:
Add optional min_length to pad_sequence

Summary: Small quality of life change: sometimes you want to pad your input to at least a certain size.

Differential Revision: D10139793

fbshipit-source-id: 544aaea511dc6aa58814ec036a793b594491a787

ezyang commented:
Sorry about the delay, but can we please have a test for this?

ezyang commented:
I generally check where the existing calls to `pad_sequence` are in `test/`. `test/test_nn.py` is probably the ticket.
